File: c-002/main.php
<?php
function fibEvenProc($max) {
    $sum = 0;
    $prev = 0;
    $next = 1;
    while ($next < $max) {
        $current = $prev + $next;
        if ($current % 2 == 0) {
            $sum += $current;
        }
        $prev = $next;
        $next = $current;
    }
    return $sum;
}

print fibEvenProc(4_000_000);

File: c-002/main.r
fibEvenProc <- function(max) {
	sum <- 0
    a <- 0
    b <- 1
	while (b < max) {
		current <- a + b
        a <- b
		b <- current
		if (current %% 2 == 0) {
			sum <- sum + current
		}
	}
	sum
}

print(fibEvenProc(4000000))
